enterprise/server/test/integration/workflow/workflow_test.go

```diff
@@ -0,0 +1,158 @@
+package workflows_test
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/test/integration/remote_execution/rbetest"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/testutil/testgit"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/workflow/service"
+	"github.com/buildbuddy-io/buildbuddy/server/backends/repo_downloader"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/app"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testbazel"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testenv"
+	"github.com/buildbuddy-io/buildbuddy/server/util/testing/flags"
+	"github.com/stretchr/testify/require"
+
+	bbspb "github.com/buildbuddy-io/buildbuddy/proto/buildbuddy_service"
+	ctxpb "github.com/buildbuddy-io/buildbuddy/proto/context"
+	inpb "github.com/buildbuddy-io/buildbuddy/proto/invocation"
+	uidpb "github.com/buildbuddy-io/buildbuddy/proto/user_id"
+	wfpb "github.com/buildbuddy-io/buildbuddy/proto/workflow"
+)
+
+var (
+	simpleRepoContents = map[string]string{
+		"WORKSPACE": `workspace(name = "test")`,
+		"BUILD": `
+sh_binary(
+    name = "nop",
+    srcs = ["nop.sh"],
+)
+`,
+		"nop.sh": ``,
+		"buildbuddy.yaml": `
+actions:
+  - name: "Test action"
+    triggers: { push: { branches: [ master ] } }
+    bazel_commands: [ "build //:nop" ]
+`,
+	}
+)
+
+func newWorkflowsTestEnv(t *testing.T, gp interfaces.GitProvider) *rbetest.Env {
+	port := app.FreePort(t)
+	// Set events_api_url to point to the test BB app server (this gets
+	// propagated to the CI runner so it knows where to publish build events).
+	flags.Set(t, "app.events_api_url", fmt.Sprintf("grpc://localhost:%d", port))
+	// Use bare execution -- Docker isn't supported in tests yet.
+	flags.Set(t, "remote_execution.workflows_default_image", "none")
+	// Use a pre-built bazel instead of invoking bazelisk, which significantly
+	// slows down the test.
+	flags.Set(t, "remote_execution.workflows_ci_runner_bazel_command", testbazel.BinaryPath(t))
+
+	env := rbetest.NewRBETestEnv(t)
+	env.AddBuildBuddyServerWithOptions(&rbetest.BuildBuddyServerOptions{
+		GRPCPort: port,
```

Comment:
Do you need this port override? Could you instead create the server and use the returned port to set the flag?